src/main/java/org/folio/fqm/utils/EntityTypeUtils.java

```diff
@@ -136,4 +131,27 @@ public static Pair<String, String> splitFieldIntoAliasAndField(String field) {
     }
     return Pair.of(field.substring(0, dotIndex), field.substring(dotIndex + 1));
   }
+
+  private static List<EntityTypeColumn> getIdColumns(EntityType entityType) {
+    List<EntityTypeColumn> idColumns = new ArrayList<>(
+      entityType.getColumns()
+        .stream()
+        .filter(column -> Boolean.TRUE.equals(column.getIsIdColumn()))
+        .toList()
+    );
+
+    // Move tenantId to the end if present (required for compatibility with bulk-edit)
+    orderIdColumns(idColumns);
+    return idColumns;
+  }
+
+  private static void orderIdColumns(List<EntityTypeColumn> idColumns) {
+    for (int i = 0; i < idColumns.size(); i++) {
+      if (idColumns.get(i).getName().contains("tenant_id")) {
+        EntityTypeColumn tenantIdColumn = idColumns.remove(i);
+        idColumns.add(tenantIdColumn);
+        break;
+      }
+    }
+  }
```

Comment:
Not saying this is a good idea, or that this code will even compile, but I'd consider doing something like this (plus comments) to make `orderIdColumns()` return a new list instead of modifying the existing one. It's slower, but less likely lead to weird bugs 5 years and a dozen refactors from now.
```suggestion
  private static List<EntityTypeColumn> orderIdColumns(List<EntityTypeColumn> idColumns) {
    return idColumns.stream()
        .sorted((a, b) -> {
            if (a.getName().equals(b.getName()) return 0;
            if (a.getName().equals("tenant_id")) return 1;
            if (b.getName().equals("tenant_id")) return -1;
            return 0;
        })
        .toList();
  }
```